pkg/virt-api/rest/subresource.go

```diff
@@ -148,7 +151,66 @@ func (app *SubresourceAPIApp) ConsoleRequestHandler(request *restful.Request, re
 	app.streamRequestHandler(request, response, CONSOLE)
 }
 
+func getChangeRequestJson(vm *v1.VirtualMachine, changes ...v1.VirtualMachineStateChangeRequest) (string, error) {
+	failOnConflict := true
+	if len(changes) == 1 && changes[0].Action == v1.StopRequest {
+		// If this is a stopRequest, replace all existing StateChangeRequests.
+		failOnConflict = false
+	}
+
+	verb := "add"
+	changeRequests := []v1.VirtualMachineStateChangeRequest{}
+	for _, change := range vm.Status.StateChangeRequests {
+		if change.Action == v1.StartRequest || change.Action == v1.StopRequest {
+			// If there are already existing start/stop requests:
+			//   new start/restart requests should fail
+			//   new stop requests should replace all existing requests.
+			if failOnConflict {
+				return "", fmt.Errorf("unable to complete request: stop/start already underway")
+			}
+		} else {
+			// preserve unknown StateChangeRequests
+			changeRequests = append(changeRequests, change)
```

Comment:
No, the loop is not pointless. It's checking for conflicts--per what we agreed on.

You could argue that the "else" is pointless, but it's future proofing this piece of code so that adding new state change request types won't suddenly break this routine. I'm not going to assume that anybody adding a new StateChangeRequest type is going to automatically know they'll need to fix that. After all, the point of turning this into a list at all was so we left that door open.